funnel/forms/proposal.py

```diff
@@ -6,36 +6,65 @@
 from baseframe.forms.sqlalchemy import QuerySelectField
 from ..models import Project, Profile
 
-__all__ = ['TransferProposal', 'ProposalForm', 'ProposalTransitionForm', 'ProposalMoveForm']
+__all__ = ['TransferProposal', 'ProposalForm', 'ProposalTransitionForm', 'ProposalLabelsForm',
+    'ProposalMoveForm', 'ProposalLabelsAdminForm']
+
+
+def proposal_label_form(project, proposal, for_admin=False):
+    """
+    Returns a label form for the given project and proposal.
+    """
+    class ProposalLabelForm(forms.Form):
+        pass
+
+    for label in project.labels:
+        field_kwargs = {
+            'label': label.form_label_text,
+            'description': label.description,
+            'validators': [forms.validators.DataRequired(__("Please select one"))] if label.required else [],
+        }
+        if label.has_options:
+            FieldType = forms.RadioField
+            field_kwargs['choices'] = [(option.name, option.title) for option in label.options if not option.archived]
+        else:
+            FieldType = forms.BooleanField
+
+        if for_admin:
```

Comment:
Do not reuse forms for different interfaces. Make these separate forms. This form is not suitable for showing on the proposal view page because there is no way to remove an option.